lib/xenops_server.ml

```diff
@@ -1680,29 +1687,49 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.task_handle) : unit
 
       (* set up the destination domain *)
       debug "VM.receive_memory creating domain and restoring VIFs";
-      (try
-         perform_atomics (
-           simplify [VM_create (id, Some memory_limit);] @
-           (* Perform as many operations as possible on the destination domain before pausing the original domain *)
-           (atomics_of_operation (VM_restore_vifs id))
-         ) t;
-         Handshake.send s Handshake.Success
-       with e ->
-         Handshake.send s (Handshake.Error (Printexc.to_string e));
-         raise e
+
+      finally (fun ()->
```

Comment:
`finally (finally () () ... )()` gets very hard to parse. Can you extract the needed functions and shink the double finally to a few lines?

Reply:
I've removed the internal finally by moving the ACK event up, so that the ACK is always executed regardless of any failure during VM_create. This simplifies the code and we don't need the 3rd commit anymore: 905f9ad882e403eda88fffe0399dd7a7b9d8078d